### ISSUE TITLE
Add ability to tag aws_launch_templates (not the created resources, the template itself)

### ISSUE THREAD

**Issue by tdmalone** [terraform-providers/terraform-provider-aws#4650]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I'd love to see the ability to manage tags on an [aws_launch_template](https://www.terraform.io/docs/providers/aws/r/launch_template.html). The ability exists to manage tags applied to created resources, but not (as far as I can tell) on the launch template itself. This ability does exist, however, through the AWS web console, so it is a resource that supports tagging.

### New or Affected Resource(s)

* [aws_launch_template](https://www.terraform.io/docs/providers/aws/r/launch_template.html)

### Potential Terraform Configuration

```hcl
resource "aws_launch_template" "foo" {
  tags {
    Name = "foo"
  }
}
```

### References

* _Semi_-related to #4649 documentation request

**Comment by bflad** [comment]
Support for a new `tags` argument in the `aws_launch_template` resource has been merged in via #4763 and will release with version 1.23.0 of the AWS provider, likely middle of next week. 🎉

**Comment by bflad** [comment]
This has been released in [version 1.23.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1230-june-14-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
